### PR TITLE
Fix data race SD-card

### PR DESCRIPTION
**Pull request by ghost**
Fix an issue with concurrent access to the sd-card, by locking the card also for ioctl-calls. This fixes a bug, that would lead to crashes and erros, if two concurrent threads access the sd-card and at least one of them uses ioctl.

### CODE REVIEW

File: subsys/sd/sd_ops.c [hunk 1]
```diff
@@ -756,6 +756,11 @@ int card_write_blocks(struct sd_card *card, const uint8_t *wbuf, uint32_t start_
 /* IO Control handler for SD MMC */
 int card_ioctl(struct sd_card *card, uint8_t cmd, void *buf)
 {
+	int ret = k_mutex_lock(&card->lock, K_MSEC(CONFIG_SD_LOCK_TIMEOUT));
+	if (ret) {
```
Comment (kica-z) [inline subsys/sd/sd_ops.c:762]: Could you add an explicit check? I know this is just my opinion but everything that is not a bool should be explicitly checked for a value.

**Review by kica-z — CHANGES_REQUESTED** [review]
Very nice find. Thanks for the fix. Just one little detail ;)

File: subsys/sd/sd_ops.c (hunk 1 above)
Comment (ghost) [reply subsys/sd/sd_ops.c:762]: I aligned it with the style of the rest of the file.

**Review by kica-z — APPROVED** [review]

File: subsys/sd/sd_ops.c (hunk 1 above)
Comment (ghost) [reply subsys/sd/sd_ops.c:762]: as discussed ;)

**Pull request merged**